raiden/transfer/channel.py

```diff
@@ -389,6 +406,146 @@ def is_valid_lockedtransfer(
+        locksroot_without_lock = merkleroot(merkletree)
+
+        (is_valid, signature_msg) = is_valid_signature(
+            received_balance_proof,
+            sender_state.address,
+        )
+
+        if not is_valid:
+            # The signature must be valid, otherwise the balance proof cannot be
+            # used onchain
+            msg = 'Invalid LockExpired message. {}'.format(signature_msg)
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.chain_id != channel_state.chain_id:
+            msg = (
+                "Invalid LockExpired message. "
+                "Chain id does not match channel's chain_id, expected: {} got: {}."
+            ).format(
+                channel_state.chain_id,
+                received_balance_proof.chain_id,
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.nonce != expected_nonce:
+            # The nonces must increase sequentially, otherwise there is a
+            # synchronization problem
+            msg = (
+                'Invalid LockExpired message. '
+                'Nonce did not change sequentially, expected: {} got: {}.'
+            ).format(
+                expected_nonce,
+                received_balance_proof.nonce,
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.locksroot != locksroot_without_lock:
+            # The locksroot must be updated to include the new lock
+            msg = (
+                'Invalid LockExpired message. '
+                "Balance proof's locksroot didn't match, expected: {} got: {}."
+            ).format(
+                hexlify(locksroot_without_lock).decode(),
+                hexlify(received_balance_proof.locksroot).decode(),
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.transferred_amount != current_transferred_amount:
+            # Given an expired lock, transferred amount should stay the same
+            msg = (
+                'Invalid LockExpired message. '
+                "Balance proof's transferred_amount changed, expected: {} got: {}."
+            ).format(
+                current_transferred_amount,
+                received_balance_proof.transferred_amount,
+            )
+
+            result = (False, msg, None)
+
+        elif not is_valid_amount(sender_state, lock.amount):
+            # We can validate that the Expired message will have a transferred amount
+            msg = (
+                "Invalid LockExpired message. "
+                "Unlocking the lock would result in an overflow. max: {} result would be: {}"
+            ).format(
+                UINT256_MAX,
+                transferred_amount_after_unlock,
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.locked_amount != expected_locked_amount:
+            # locked amount should be be the same as BP given the current lock expiry
```

Comment:
Comment is not readable. Perhaps just `locked amount should be the same found inside the balance proof?`